src/overlays/actors/ovl_Door_Ana/z_door_ana.c

```diff
@@ -21,20 +24,166 @@ const ActorInit Door_Ana_InitVars = {
     (ActorFunc)DoorAna_Update,
     (ActorFunc)DoorAna_Draw
 };
-*/
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Door_Ana_0x808E01A0/func_808E01A0.asm")
+static ColliderCylinderInit sCylinderInit = { 
+    { COLTYPE_NONE, AT_NONE, AC_ON | AC_TYPE_PLAYER, OC1_NONE, OC2_NONE, COLSHAPE_CYLINDER, },
+    { ELEMTYPE_UNK2, { 0x00000000, 0x00, 0x00 }, { 0x00000008, 0x00, 0x00 }, TOUCH_NONE | TOUCH_SFX_NORMAL, BUMP_ON, OCELEM_NONE, },
+    { 50, 10, 0, { 0, 0, 0 } },
+};
+
+// entrances grottos can link to statically, type DOORANA_TYPE_ADJACENT uses scene exit addresses instead
+static u16 entrances[] = {
+    0x1A00, 0x1400, 0x1410, 0x1420, 0x1430, 0x1440, 0x1450, 0x1460,
+    0x1470, 0x1480, 0x1490, 0x14A0, 0x14B0, 0x14C0, 0x14D0, 0x0000, 0x0000, 0x0000,
+};
+
+void DoorAna_SetupAction(DoorAna* this, DoorAnaActionFunc actionFunction) {
+    this->actionFunc = actionFunction;
+}
+
+void DoorAna_Init(Actor* thisx, GlobalContext* globalCtx) {
+    DoorAna* this = THIS;
+    s32 grottoType = GET_DOORANA_TYPE(this);
+    this->actor.shape.rot.y = this->actor.shape.rot.z = 0;
+
+    if ((grottoType == DOORANA_TYPE_UNK) || (grottoType == DOORANA_TYPE_HIDDEN)) {
+        if (grottoType == DOORANA_TYPE_HIDDEN) {
+            Collider_InitAndSetCylinder(globalCtx, &this->bombCollider, &this->actor, &sCylinderInit);
+
+        } else {
+            this->actor.flags |= 0x10; // always update 
+        }
+
+        Actor_SetScale(&this->actor, 0);
+        DoorAna_SetupAction(this, DoorAna_WaitClosed);
+
+    } else {
+        DoorAna_SetupAction(this, DoorAna_WaitOpen);
+    }
+
+    this->actor.targetMode = 0;
+}
+
+void DoorAna_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    DoorAna* this = THIS;
+    s32 grottoType = GET_DOORANA_TYPE(this);
+
+    if (grottoType == DOORANA_TYPE_HIDDEN) {
+        Collider_DestroyCylinder(globalCtx, &this->bombCollider);
+    }
+}
+
+void DoorAna_WaitClosed(DoorAna* this, GlobalContext* globalCtx) {
+    s32 grottoIsOpen = 0;
+    u32 grottoType = GET_DOORANA_TYPE(this);
+
+    if ((grottoType) == DOORANA_TYPE_UNK) {
+        // in OOT decomp its marked as open with storms, but does not seem to open with storms in MM
+        if ((this->actor.xyzDistToPlayerSq < 40000.0f) && (func_800F13E8(globalCtx, 5) != 0)) {
```

Comment:
For (x != 0) type stuff I've been told to just omit the != 0 if it still matches, not sure about the current thoughts on that